src/synthesizer/mod.rs

```diff
@@ -0,0 +1,335 @@
+use crate::ir::conditions::ConditionIr;
+use crate::ir::mappings::MappingInstruction;
+use crate::ir::reference::Origin;
+use crate::ir::reference::PseudoParameter;
+use crate::ir::CloudformationProgramIr;
+use crate::parser::lookup_table::MappingInnerValue;
+use crate::parser::resource::ResourceValue;
+use crate::parser::sub::{sub_parse_tree, SubValue};
+use crate::CloudformationParseTree;
+use std::collections::HashMap;
+use voca_rs::case::camel_case;
+
+// TODO - remove parse_tree
+pub fn synthesize(parse_tree: CloudformationParseTree, ir: CloudformationProgramIr) {
```

Comment:
Can we rename this file to typescript_synthesizer?

Reply:
Sure!